app/models/account.rb

```diff
@@ -38,60 +38,82 @@ def price_groups
     end
   end
 
-  def add_or_update_member(user, new_role, session_user)
-    Account.transaction do
-      # expire old owner if new
-      if new_role == AccountUser::ACCOUNT_OWNER
-        # expire old owner record
-        @old_owner = self.owner
-        if @old_owner
-          @old_owner.deleted_at = Time.zone.now
-          @old_owner.deleted_by = session_user.id
-          @old_owner.save!
-        end
-      end
+  # Given an account_type return a param string that can be used as a variable
+  # name, filename, param, etc. Replaces any backslashes with underscore to
+  # support namespaced classes.
+  def self.account_type_to_param(account_type)
+    account_type.to_s.underscore.gsub("/", "_")
+  end
 
-      # find non-deleted record for this user and account or init new one
-      # deleted_at MUST be nil to preserve existing audit trail
-      @account_user = AccountUser.find_or_initialize_by_account_id_and_user_id_and_deleted_at(
-        self.id,
-        user.id,
-        nil
-      )
-      # set (new?) role
-      @account_user.user_role = new_role
-      # set creation information
-      @account_user.created_by = session_user.id
+  # Returns an array of subclassed Account objects.
+  # Engines can concat to this on initialization.
+  def self.account_types
+    @@account_types ||= [NufsAccount]
+  end
 
-      self.account_users << @account_user
+  # Returns an array of subclassed Account objects available across multiple
+  # facilities. Engines can concat to this on initialization.
+  def self.global_account_types
+    @@global_account_types ||= [NufsAccount]
+  end
 
-      raise ActiveRecord::Rollback unless self.save
-    end
+  # Returns an array of subclassed Account objects available per facility.
+  # Engines can concat to this on initialization.
+  def self.facility_account_types
+    @@facility_account_types ||= []
+  end
 
-    return @account_user
+  # Returns an array of subclassed Account objects availalbe for statements.
+  def self.statement_account_types
+    facility_account_types
   end
```

Comment:
Good idea.  I added `AccountConfig` instead since `Account::Config` raised some rails deprecation errors.
```
/.../nucore/nucore-open/app/models/account.rb:47:in `config': Use RbConfig instead of obsolete and deprecated Config.
/.../nucore/nucore-open/app/models/account.rb:47: warning: toplevel constant Config referenced by Account::Config
/.../nucore/nucore-open/app/models/account.rb:47:in `config': Use RbConfig instead of obsolete and deprecated Config.
/.../.rbenv/versions/2.1.2/lib/ruby/2.1.0/rbconfig/obsolete.rb:20:in `method_missing': undefined method `new' for RbConfig:Module (NoMethodError)
	from /.../nucore/nucore-open/app/models/account.rb:47:in `config'
```